pkgs/development/compilers/blueprint/default.nix

```
@@ -0,0 +1,46 @@
+{ python3
+, stdenv
+, fetchFromGitLab
+, gobject-introspection
+, lib
+, meson
+, ninja
+}:
+
+stdenv.mkDerivation rec {
+  pname = "blueprint-compiler";
+  version = "unstable-2022-05-27";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "jwestman";
+    repo = pname;
+    rev = "cebd9ecadc53790cd547392899589dd5de0ac552";
+    sha256 = "sha256-mNR0ooJSRBIXy2E4avXYEdO1aSST+j41TsVg8+kitwo=";
+  };
+
+  # Requires pythonfuzz, which I've found difficult to package
+  doCheck = false;
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    python3.pkgs.wrapPython
+  ];
+  propagatedBuildInputs = [
+    gobject-introspection
+    python3
+  ];
```

Then it should at least be commented on. And Python probably does not need to be propagated.

```suggestion
  ];

  buildInputs = [
    python3
  ];

  propagatedBuildInputs = [
    # So that the compiler can find GIR files.
    gobject-introspection.setupHook
  ];
```

---

Only adding `gobject-introspection.setupHook` to propagatedBuildInputs isn't enough.

It can't find ui files `
../data/resources/resources.gresource.xml: Failed to locate ?ui/shortcuts.ui? in any source directory.
`

---

I do not think that is related to gobject-introspection. It looks to me like a failure in `glib-compile-resources`, which is probably caused by blueprint-compiler not creating the `ui` file. Perhaps there is an error earlier in the log? Do you have a branch where I could reproduce this.

---

I have a nixpkgs branch with my gtk app, which uses blueprint-compiler: https://github.com/ranfdev/nixpkgs/tree/geopard.

You can run `nix shell .#geopard` to build it.

---

The actual error is the following:

```
[3/5] Generating data/resources/blueprints with a custom command
Traceback (most recent call last):
  File "/nix/store/nfzjvw5bdk4lmi1591nck241i9xx10sq-blueprint-compiler-unstable-2022-05-27/lib/python3.9/site-packages/blueprintcompiler/gir.py", line 411, in __init__
    self.includes = { include["name"]: get_namespace(include["name"], include["version"]) for include in xml.get_elements("include") }
  File "/nix/store/nfzjvw5bdk4lmi1591nck241i9xx10sq-blueprint-compiler-unstable-2022-05-27/lib/python3.9/site-packages/blueprintcompiler/gir.py", line 411, in <dictcomp>
    self.includes = { include["name"]: get_namespace(include["name"], include["version"]) for include in xml.get_elements("include") }
  File "/nix/store/nfzjvw5bdk4lmi1591nck241i9xx10sq-blueprint-compiler-unstable-2022-05-27/lib/python3.9/site-packages/blueprintcompiler/gir.py", line 53, in get_namespace
    raise CompileError(f"Namespace {namespace}-{version} could not be found")
blueprintcompiler.errors.CompileError: Namespace Gio-2.0 could not be found
```

So it actually also depends on introspection files for Glib/Gio, which are shipped with g-i.